### PR TITLE
Build both intel and arm docker images

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
`cimg/rust` isn't currently being built for arm64, so I've used [`arm64v8/rust`](https://hub.docker.com/r/arm64v8/rust) and [`rust`](https://hub.docker.com/_/rust) correspondingly, but they don't have a dedicated user specified at least.

If the approach is viable otherwise, should I set this user in the Dockerfile?

### CODE REVIEW

File: rust/1.58.0/Dockerfile [hunk 1]
```diff
@@ -1,80 +0,0 @@
-ARG RUST_VERSION=1.58.0
```
Comment (LizardWizzard) [inline rust/1.58.0/Dockerfile:1]: Could you please elaborate why this Dockerfile for 1.58.0 is removed?

Comment (LizardWizzard) [inline rust/1.58.0/Dockerfile:1]: Could you please elaborate why this Dockerfile for 1.58.0 is removed?

**Comment by cicdteam** [comment]
@SomeoneToIgnore 

my opinion:

I disagree the proposed way for intel/arm docker images. I prefer utilize multi platform build without additional suffixes (so, image name with tag will be the same for both platforms and docker hub will manage what image should be pulled)

Second point - changing base image will broke circleci CICD flows (as native rust images not adapted for it).

So, I suggest closing this PR and wait a bit, I have a plan solve multi platform issue soon (I did it already for zenith repository but not merged as build was too slow).

PS: about milti platform builds and cross-complitauin within docker:
- https://docs.docker.com/desktop/multi-arch/
- https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

File: rust/1.58.0/Dockerfile (hunk 1 above)
Comment (SomeoneToIgnore) [reply rust/1.58.0/Dockerfile:1]: This is a build argument, both files were almost identical, with the same build arguments, but their default values being different: so I've collapsed files to one and specified on extra build arg in the build scripts.

**Comment by SomeoneToIgnore** [comment]
Perfect, thanks for reading.

I'd like to try working on that when my engineer week comes (late March), but you're welcome to implement it before that 🙂

**Pull request closed without merging**